app/livechat/server/lib/Contacts.js

```diff
@@ -60,6 +64,15 @@ export const Contacts = {
 
 		LivechatVisitors.updateById(contactId, updateUser);
 
+		const room = LivechatRooms.findOneByVisitorToken(token, { _id: 1 });
```

Comment:
This doesn't make sense, this method is returning a random room, you can't make sure this is the room that needs to be updated in case the contact has more than just one room stored on the database.

My Suggestion: 
- Fetch all rooms associated with the same contact and update all of them.